content/post/2018-04-02-sct-encoding.markdown

```diff
@@ -0,0 +1,439 @@
+---
+author: Jacob Hoffman-Andrews, Boulder developer
+date: 2018-04-02T00:00:00Z
+excerpt: How Signed Certificate Timestamps get embedded in certificates
+title: "Engineering deep dive: Encoding of SCTs in certificates"
+slug: sct-encoding
+---
+
+Let's Encrypt recently [launched SCT embedding in
+certificates](https://community.letsencrypt.org/t/signed-certificate-timestamps-embedded-in-certificates/57187).
+This feature allows browsers to check that a certificate was submitted to a
+Certificate Transparency log. As part of the launch, we did a thorough review
+that the encoding of Signed Certificate Timestamps (SCTs) in our certificates
+matches the relevant specifications. In this post, I'll dive into the details.
+You'll learn more about X.509, ASN.1, DER, and TLS encoding, with references to
+the relevant RFCs.
+
+Certificate Transparency offers three ways to deliver SCTs to a browser: In a
+TLS extension, in stapled OCSP, or embedded in a certificate. We chose to
+implement the embedding method because it would just work for Let's Encrypt
+subscribers without additional work. In the SCT embedding method, we submit
+a "precertificate" with a [poison extension](#poison) to a set of
+CT logs, and get back SCTs. We then issue a real certificate based on the
+precertificate, with two changes: The poison extension is removed, and a list
+of SCTs is added in another extension.
+
+Given a certificate, let's first look for the extension. According to CT ([RFC 6962
```

Comment:
"let's first look for the extension" - since above you talk about both a poison extension and the SCT extension it might be worth being explicit here and saying "Let's first look for the SCT list extension" or similar.